src/utils.rs

```diff
@@ -177,6 +177,81 @@ pub fn canonicalize(path: impl AsRef<Path>) -> Result<PathBuf, SolcIoError> {
     res.map_err(|err| SolcIoError::new(err, path))
 }
 
+/// Returns a normalized Solidity file path for the given import path based on the specified
+/// directory.
+///
+/// This function resolves `./` and `../`, but, unlike [`canonicalize`], it does not resolve
+/// symbolic links.
+///
+/// The function returns an error if the normalized path does not exist in the file system.
+///
+/// See also: <https://docs.soliditylang.org/en/v0.8.23/path-resolution.html>
+pub fn normalize_solidity_import_path(
+    directory: impl AsRef<Path>,
+    import_path: impl AsRef<Path>,
+) -> Result<PathBuf, SolcIoError> {
+    let original = directory.as_ref().join(import_path);
+    let cleaned = clean_solidity_path(&original);
+
+    // this is to align the behavior with `canonicalize`
+    use path_slash::PathExt;
+    let normalized = PathBuf::from(dunce::simplified(&cleaned).to_slash_lossy().as_ref());
+
+    fs::metadata(&normalized).map(|_| normalized).map_err(|err| SolcIoError::new(err, original))
```

Comment:
what's this call for?

Reply:
The function `normalize_solidity_import_path` checks if the returned path exists in the file system. The use of `fs::metadata(&normalized)` is to obtain an `io::Error` for non-existing files without reading their entire contents. To return the same type of error as `utils::canonicalize`, an `io::Error` is necessary for `SolcIoError`.

While there is a specific function, `Path::is_file()`, that confirms if a path exists in the file system, it only returns a bool.

https://doc.rust-lang.org/std/path/struct.Path.html#method.is_file

> This is a convenience function that coerces errors to false. If you want to check errors, call `fs::metadata` and handle its `Result`.

However, I have now realized that there's an alias `Path::metadata()` for paths. Therefore, I will switch to using this function and add a comment.

https://doc.rust-lang.org/std/path/struct.Path.html#method.metadata